### ISSUE TITLE
Switch to pytest/unittest/nose2 from nose

### ISSUE THREAD

**Issue by paolostivanin** [python-zk/kazoo#597]
Hello,
just a heads-up that nose won't work anymore starting from python 3.9/3.10 :)

**Comment by jeffwidman** [comment]
Thanks for the heads up!

Closing as switching to pytest is already tracked under https://github.com/python-zk/kazoo/issues/589... it's really just waiting on someone to put up a PR
